agent/engine/task_manager.go

```diff
@@ -129,9 +133,15 @@ func (mtask *managedTask) overseeTask() {
 			// able to move some containers along.
 			llog.Debug("Task not steady state or terminal; progressing it")
 
-			// TODO:
-			// 1) Add platform resources setup and failure handling
-			// 2) Add new task resources provisioned state ?
+			// TODO: Add new resource provisioned state ?
```

Comment:
Yeah, we'll need to do something.  Right now, this will run on every task transition, which isn't what we want.

Reply:
Yes, but we don't recreate the `cgroup` if it already exists.

Reply:
I'm working on expressing transition dependencies a bit more explicitly right now (PR coming either today or tomorrow).  Rather than adding a new state, we might just want to add an explicit model for the resource to the task, and then make each of the containers dependent on that resource.

Reply:
Yep, that makes sense but I'll leave the `TODO` as is for now and we can update it following your changes

Reply:
Check out https://github.com/aws/amazon-ecs-agent/pull/989 and see if that makes sense to you as a possible way to address this.

Reply:
@samuelkarp I've looked at #989 and will update this `TODO` based off it when its been merged.